beacon_node/store/src/hot_cold_store.rs

```diff
@@ -948,6 +948,10 @@ impl<E: EthSpec, Hot: ItemStore<E>, Cold: ItemStore<E>> HotColdDB<E, Hot, Cold>
         store_updated_vector(StateRoots, db, state, &self.spec, ops)?;
         store_updated_vector(HistoricalRoots, db, state, &self.spec, ops)?;
         store_updated_vector(RandaoMixes, db, state, &self.spec, ops)?;
+        match state.fork_name(&self.spec)? {
```

Comment:
I'm not sure exactly how this would fail but I think this is necessary? Maybe `field!` could be adjusted to accept fork information for `update_pattern`?

Similarly should we stop writing historical roots to the store at Capella?

Reply:
Good idea, I took this idea and ran with it in https://github.com/sigp/lighthouse/pull/3865/commits/a435ae7ea9291f0ba0011cc9343845686fb8c2d8 which I've pushed to this branch.

Capella `beacon-chain` tests are passing now. I'll try a local Capella testnet and then we can merge :rocket: